CMakeLists.txt

```diff
@@ -29,7 +29,16 @@ endif()
 mason_use(catch VERSION 1.9.6 HEADER_ONLY)
 include_directories(SYSTEM ${MASON_PACKAGE_catch_INCLUDE_DIRS})
 
+mason_use(benchmark VERSION 1.2.0)
+include_directories(SYSTEM ${MASON_PACKAGE_benchmark_INCLUDE_DIRS})
+
 include_directories("${PROJECT_SOURCE_DIR}/include")
 
 file(GLOB TEST_SOURCES test/*.cpp)
 add_executable(unit-tests ${TEST_SOURCES})
+
+file(GLOB BENCH_SOURCES bench/*.cpp)
+add_executable(bench-tests ${BENCH_SOURCES})
```

Comment:
Just realized that `libbenchmark.a` supports threads and therefore needs pthread support. Because the way to enable pthreads in your build is not consistent across platforms, cmake has a way to do this for us. Just add:

```
find_package(Threads REQUIRED)
```

Somewhere above `add_executable`. And then add `${CMAKE_THREAD_LIBS_INIT}` like:

```
target_link_libraries(bench-tests ${MASON_PACKAGE_benchmark_STATIC_LIBS} ${CMAKE_THREAD_LIBS_INIT})
```